contracts/test/TokenVestingTest.t.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0;
+
+import "./config/BaseTest.t.sol";
+import "../TokenVesting.sol";
+
+contract TokenVestingTest is BaseTest {
+    TokenVesting public tokenVesting;
+    address alice = vm.addr(1);
+    address bob = vm.addr(2);
+    
+    struct Vest {
+        uint256 total;
+        uint256 claimedAmount;
+        bool isClaimed;
+    }
+
+    function setUp() public {
+        tokenVesting = new TokenVesting();
+    }
+
+    function test_settingVestingAmounts() public {
+        address[] memory addresses = new address[](2);
+        addresses[0] = alice;
+        addresses[1] = bob;
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = 700;
+        amounts[1] = 300;
+
+        vm.prank(tokenVesting.owner()); 
+        tokenVesting.setVestingAmounts(1000, addresses, amounts);
+
+        assertEq(tokenVesting.getVestingAmount(alice), 700);
+        assertEq(tokenVesting.getVestingAmount(bob), 300);
+    }
+
+    function testFail_nonOwnerSettingVestingAmounts() public {
```

Comment:
Can do this as a fuzz test actually: https://book.getfoundry.sh/forge/fuzz-testing